subprojects/docs/src/docs/userguide/taskConfiguration.adoc

```diff
@@ -0,0 +1,269 @@
+- Migrate tasks that are configured by name. 
+- Migrate tasks created in custom plugins to use TODO `register(...)` and `configureEach {}`.
+- Use the information about which task types are created or use the old API to guide which build scripts or plugins to modify. 
+- When inside a task configuration action, avoid configuring anything other than that task.  This includes other tasks and extensions. 
+
+[[sec:old_vs_new_configuration_api_overview]]
+=== Old vs New API overview
+
+[NOTE]
+====
+Methods that take a `groovy.lang.Closure` are covered in the new API with methods taking `org.gradle.api.Action`.
+====
+
+[cols="a,a", options="header"]
+|===
+| Old API | New API
+
+| api:org.gradle.api.tasks.TaskContainer#create(java.util.Map)[]
+| No direct equivalent.
+2+| Use one of the alternatives below.
+
+| api:org.gradle.api.tasks.TaskContainer#create(java.util.Map,groovy.lang.Closure)[]
+| No direct equivalent.
+2+| Use one of the alternatives below.
+
+| api:org.gradle.api.tasks.TaskContainer#create(java.lang.String)[]
+| api:org.gradle.api.tasks.TaskContainer#register(java.lang.String)[]
+2+| This returns a `TaskProvider` instead of a `Task`.
+
+| api:org.gradle.api.tasks.TaskContainer#create(java.lang.String,groovy.lang.Closure)[]
+| api:org.gradle.api.tasks.TaskContainer#register(java.lang.String,org.gradle.api.Action)[]
+2+| This returns a `TaskProvider` instead of a `Task`.
+
+| api:org.gradle.api.tasks.TaskContainer#create(java.lang.String,java.lang.Class)[]
+| api:org.gradle.api.tasks.TaskContainer#register(java.lang.String,java.lang.Class)[]
+2+| This returns a `TaskProvider` instead of a `Task`.
+
+| api:org.gradle.api.tasks.TaskContainer#create(java.lang.String,java.lang.Class,org.gradle.api.Action)[]
+| api:org.gradle.api.tasks.TaskContainer#register(java.lang.String,java.lang.Class,org.gradle.api.Action)[]
+2+| This returns a `TaskProvider` instead of a `Task`.
+
+| api:org.gradle.api.tasks.TaskContainer#create(java.lang.String,java.lang.Class,java.lang.Object...)[]
+| api:org.gradle.api.tasks.TaskContainer#register(java.lang.String,java.lang.Class,java.lang.Object...)[]
+2+| This returns a `TaskProvider` instead of a `Task`.
+
+| api:org.gradle.api.tasks.TaskCollection#getByName(java.lang.String)[]
+| api:org.gradle.api.tasks.TaskCollection#named(java.lang.String)[]
+2+| This returns a `TaskProvider` instead of a `Task`.
+
+| api:org.gradle.api.tasks.TaskCollection#getByName(java.lang.String,groovy.lang.Closure)[]
+| `named(java.lang.String).configure(Action)`
+2+| This returns `void`, so it cannot be chained.
+
+| api:org.gradle.api.tasks.TaskContainer#getByPath(java.lang.String)[]
+| No direct equivalent.
+2+| Accessing tasks from another project requires a specific ordering of project evaluation. 
+
+| api:org.gradle.api.NamedDomainObjectCollection#findByName(java.lang.String)[]
+| No direct equivalent.
```

Comment:
We might point out that some of these equivalents are missing and we may add them later, and that some methods will just go away. For example, we might add something like `findByName()` or `maybeCreate()` later but we might want to kill off `getByPath()` (and replace it with something that returns a `TaskReference`).

Or maybe we just point out that we'll probably add some more convenience methods later, based on people's feedback and that this is just the first round with the minimal set of new methods.